drivers/display/display_ili9340.c

```diff
@@ -1,154 +1,152 @@
 /*
  * Copyright (c) 2017 Jan Van Winkel <jan.van_winkel@dxplore.eu>
  * Copyright (c) 2019 Nordic Semiconductor ASA
+ * Copyright (c) 2020 Pavlo Hamov <pasha.gamov@gmail.com>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include "display_ili9340.h"
 #include <drivers/display.h>
 
-#define LOG_LEVEL CONFIG_DISPLAY_LOG_LEVEL
-#include <logging/log.h>
-LOG_MODULE_REGISTER(display_ili9340);
-
 #include <drivers/gpio.h>
 #include <sys/byteorder.h>
 #include <drivers/spi.h>
 #include <string.h>
 
-struct ili9340_data {
-#ifdef DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_CONTROLLER
-	struct device *reset_gpio;
-#endif
-	struct device *command_data_gpio;
-	struct device *spi_dev;
-	struct spi_config spi_config;
-#ifdef DT_INST_0_ILITEK_ILI9340_CS_GPIOS_CONTROLLER
-	struct spi_cs_control cs_ctrl;
-#endif
-};
-
-#define ILI9340_CMD_DATA_PIN_COMMAND 1
-#define ILI9340_CMD_DATA_PIN_DATA 0
+#include <logging/log.h>
+LOG_MODULE_REGISTER(display_ili9340, CONFIG_DISPLAY_LOG_LEVEL);
 
-/* The number of bytes taken by a RGB pixel */
-#ifdef CONFIG_ILI9340_RGB565
-#define ILI9340_RGB_SIZE 2U
+#if (DT_INST_0_ILITEK_ILI9XXX_TYPE_ENUM == 0)
+#define ILI9XXX_X_MAX 132U
+#define ILI9XXX_Y_MAX 162U
 #else
-#define ILI9340_RGB_SIZE 3U
+#define ILI9XXX_X_MAX 240U
+#define ILI9XXX_Y_MAX 320U
 #endif
 
-static void ili9340_exit_sleep(struct ili9340_data *data)
-{
-	ili9340_transmit(data, ILI9340_CMD_EXIT_SLEEP, NULL, 0);
-	k_sleep(K_MSEC(120));
-}
-
-static int ili9340_init(struct device *dev)
-{
-	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+#define RES_VER DT_INST_0_ILITEK_ILI9XXX_HEIGHT
+#define RES_HOR DT_INST_0_ILITEK_ILI9XXX_WIDTH
 
-	LOG_DBG("Initializing display driver");
+#define DEV_TYPE DT_INST_0_ILITEK_ILI9XXX_TYPE
 
-	data->spi_dev = device_get_binding(DT_INST_0_ILITEK_ILI9340_BUS_NAME);
-	if (data->spi_dev == NULL) {
-		LOG_ERR("Could not get SPI device for ILI9340");
-		return -EPERM;
-	}
+#if (RES_VER > ILI9XXX_Y_MAX) && (RES_VER > ILI9XXX_X_MAX)
+#error "Please check height of \"ilitek,"DEV_TYPE""\" device"
```

Comment:
Better be consistent in error messages; `"ilitek,"DEV_TYPE"` VS `ilitek,ili9340` bellow